### PR TITLE
RET-3994  Idam and CCD integration

### PR DESCRIPTION
**Pull request by brendanmchugh**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RET-3993
https://tools.hmcts.net/jira/browse/RET-3994

### Change description ###

This PR adds integrations with:
- CCD
- IDAM

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: charts/et-hearings-api/values.preview.template.yaml [hunk 1]
```diff
@@ -2,3 +2,11 @@ java:
   # Don't modify below here
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
+  keyVaults:
+  #  et-hearings-api:
+  #    secrets:
+  #      - name: et-hearings-api-s2s-secret
+  #        alias: ET_HEARINGS_S2S_SECRET
```
Comment (JackReevies) [inline charts/et-hearings-api/values.preview.template.yaml:9]: Is this needed at all if its declared in `values.yml` too?

File: config/owasp/suppressions.xml [hunk 2]
```diff
@@ -29,5 +29,9 @@
     <notes>jackson.core/jackson-databind@2.15.0</notes>
     <cve>CVE-2023-35116</cve>
   </suppress>
+  <suppress>
+    <notes>bcprov-jdk18on-1.73.jar Bouncy Castle Crypto package</notes>
+    <cve>CVE-2023-33201</cve>
+  </suppress>
```
Comment (JackReevies) [inline config/owasp/suppressions.xml:35]: Have you tried updating to 1.76? Apparently this version fixes this CVE

File: src/main/java/uk/gov/hmcts/reform/et/helper/mapping/ServiceHearingValuesMapping.java [hunk 3]
```diff
@@ -12,41 +12,43 @@ private ServiceHearingValuesMapping() {
...
+        .judiciary(null)
+        .hearingIsLinkedFlag(false)
+        .parties(null)
+        .caseFlags(null)
+        .hmctsServiceID(null)
+        .hearingChannels(null)
+        .screenFlow(null)
+        .vocabulary(null)
+        .caseInterpreterRequiredFlag(false)
+        .build();
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/reform/et/helper/mapping/ServiceHearingValuesMapping.java:51]: These should be indented as they were before

File: src/main/java/uk/gov/hmcts/reform/et/service/VerifyTokenService.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.et.service;
+
+
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/reform/et/service/VerifyTokenService.java:3]: double blank line?

**Review by JackReevies — CHANGES_REQUESTED** [review]

File: charts/et-hearings-api/values.preview.template.yaml (hunk 1 above)
Comment (brendanmchugh) [reply charts/et-hearings-api/values.preview.template.yaml]: removed those commented out lines

File: config/owasp/suppressions.xml (hunk 2 above)
Comment (brendanmchugh) [reply config/owasp/suppressions.xml]: add a dependency to 1.76 and removed the CVE suppression.

File: src/main/java/uk/gov/hmcts/reform/et/service/VerifyTokenService.java (hunk 4 above)
Comment (brendanmchugh) [reply src/main/java/uk/gov/hmcts/reform/et/service/VerifyTokenService.java]: removed

**Review by dharmendrak — APPROVED** [review]
LGTM

**Review by JackReevies — APPROVED** [review]

**Pull request merged**